src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts

```diff
@@ -47,26 +54,31 @@ function isNavDirectLink(value: NavNode): value is NavDirectLink {
 	return value.hasOwnProperty('href')
 }
 
+let TUTORIAL_MAP
+
 /**
  * Prepares all sidebar nav items for client-side rendering. Keeps track of the
  * index of each node using `startingIndex` and the `traversedNodes` property
  * returned from `prepareNavNodeForClient`. Also returns its own
  * `traversedNodes` since it is recursively called in `prepareNavDataForClient`.
  */
-function prepareNavDataForClient({
+async function prepareNavDataForClient({
 	basePaths,
 	nodes,
 	startingIndex = 0,
 }: {
 	basePaths: string[]
 	nodes: NavNode[]
 	startingIndex?: number
-}): { preparedItems: MenuItem[]; traversedNodes: number } {
+}): Promise<{ preparedItems: MenuItem[]; traversedNodes: number }> {
 	const preparedNodes = []
 
+	TUTORIAL_MAP = await getTutorialMap()
```

Comment:
```suggestion
	TUTORIAL_MAP = TUTORIAL_MAP ?? await getTutorialMap()
```
Could utilize the closure?

Reply:
Good call!